Add files via upload

KMironova opened the pull request:
Finished API homework_1

KMironova committed:
Add files via upload

alla-blinkova commented:
Tests don't contain preconditions and postconditions (see 2.2 in the task description)

alla-blinkova commented:
"ChekFailWithoutRequiredAttributeTest" - the test fails, but it looks like it shouldn't as it is a correct negative test

alla-blinkova commented:
"GroovyDataProviderTest": 

- [x] Why do you need Property Transfer step in this test? 
- [ ] The test doesn't check the correct spell in the response, please add corresponding assertion.
- [ ] Are you sure you need to store testSize in properties?

alla-blinkova reviewed (changes requested):
Changes according to the comments are required

KMironova commented:
> "GroovyDataProviderTest":
> 
> * [x]  The test doesn't check the correct spell in the response, please add corresponding assertion.

Correct spell chek is occur in CheckSpellCorrectForText

KMironova commented:
> "GroovyDataProviderTest":
> 
> * [x]  Are you sure you need to store testSize in properties?

Yes, because I need to transfer this value from one groove script to another and I decide to do it with property value

KMironova commented:
> "GroovyDataProviderTest":
> 
> * [x]  Why do you need Property Transfer step in this test?

I need this tool to transfer value (that contains words for spell chek) to propertie and to use as many different tools as possible in different variations

KMironova committed:
Delete homework-8-soapui-project.xml

KMironova committed:
Add files via upload

1) In CheckNotFixForRusTextWithEnAttributeTest added assertion check to fail
2) Rename ChekFailWithoutRequiredAttributeTest to CheckIgnoreCaseTest and added relevant steps to it. This was done because the SOAP UI mechanic does not allow the use of special characters in the prologue for validation.
3) Added preconditions and postconditions in GroovyTestWithPropertiesTest

alla-blinkova commented:
> Tests don't contain preconditions and postconditions (see 2.2 in the task description)

Couldn't find preconditions and postconditions in tests or suites, please add some

alla-blinkova commented:
> > "GroovyDataProviderTest":
> > 
> > * [x]   The test doesn't check the correct spell in the response, please add corresponding assertion.
> 
> Correct spell chek is occur in CheckSpellCorrectForText

Please add check for the expected spell, now I can see only simple assertions. I suppose you need to use an array for expected words in the script, two-dimensional array is better.
![image](https://user-images.githubusercontent.com/26868439/133978979-f3bbf02e-6602-4b5d-9196-86760d6c605e.png)

alla-blinkova commented:
> > "GroovyDataProviderTest":
> > 
> > * [x]   Are you sure you need to store testSize in properties?
> 
> Yes, because I need to transfer this value from one groove script to another and I decide to do it with property value

It looks like counter works wrong, test is not executed for the last word in array.

alla-blinkova commented:
One more thing about "GroovyDataProviderTest" - could you remove Property Transfer step from this test to make it simpler?

KMironova committed:
Delete homework-8-soapui-project.xml

KMironova committed:
Add files via upload

1) add setup and tear down to GroovyTestWithPropertiesTest
2) fix GroovyDataProviderTest

alla-blinkova commented:
"GroovyDataProviderTest" - if one assertion fails (e.g. 1st in the data set), the whole test fails and stops immediately, but it should not (see 2.5 in the task description)

alla-blinkova reviewed (changes requested):
"GroovyDataProviderTest" - if one assertion fails (e.g. 1st in the data set), the whole test fails and stops immediately, but it should not (see 2.5 in the task description)

KMironova committed:
Delete homework-8-soapui-project.xml

KMironova committed:
Add files via upload
